packages/fuels-types/src/param_types.rs

```diff
@@ -1215,4 +1249,84 @@ mod tests {
+        assert!(ParamType::B256.contains_no_nested_vectors());
+        assert!(ParamType::String(10).contains_no_nested_vectors());
+        assert!(ParamType::RawSlice.contains_no_nested_vectors());
+        Ok(())
+    }
+
+    #[test]
+    fn contains_no_nested_vectors_complex_types() -> Result<()> {
+        let base_vector = ParamType::Vector(Box::from(ParamType::U8));
+        let tuples_no_nested_vec = vec![
+            ("Bim".to_string(), ParamType::U16),
+            ("Bam".to_string(), ParamType::Bool),
+        ];
+        let tuples_with_nested_vec = vec![
+            ("Zim".to_string(), ParamType::U64),
+            ("Zam".to_string(), ParamType::U32),
+            ("Boum".to_string(), base_vector.clone()),
+        ];
+        let param_types_no_nested_vec = vec![ParamType::U64, ParamType::U32];
+        let param_types_nested_vec = vec![ParamType::Unit, ParamType::Byte, base_vector.clone()];
+
+        assert!(base_vector.contains_no_nested_vectors());
+        assert!(!ParamType::Vector(Box::from(base_vector.clone())).contains_no_nested_vectors());
+
+        assert!(ParamType::Array(Box::from(ParamType::U8), 10).contains_no_nested_vectors());
+        assert!(!ParamType::Array(Box::from(base_vector), 10).contains_no_nested_vectors());
+
+        assert!(ParamType::Tuple(param_types_no_nested_vec.clone()).contains_no_nested_vectors());
+        assert!(!ParamType::Tuple(param_types_nested_vec.clone()).contains_no_nested_vectors());
+
+        assert!(ParamType::Struct {
+            name: "StructName".to_string(),
+            generics: param_types_no_nested_vec.clone(),
+            fields: tuples_no_nested_vec.clone(),
+        }
+        .contains_no_nested_vectors());
+        assert!(!ParamType::Struct {
+            name: "StructName".to_string(),
+            generics: param_types_nested_vec.clone(),
+            fields: tuples_no_nested_vec.clone()
+        }
+        .contains_no_nested_vectors());
+        assert!(!ParamType::Struct {
+            name: "StructName".to_string(),
+            generics: param_types_no_nested_vec.clone(),
+            fields: tuples_with_nested_vec.clone()
+        }
+        .contains_no_nested_vectors());
+
+        assert!(ParamType::Enum {
+            name: "EnumName".to_string(),
+            variants: EnumVariants::new(tuples_no_nested_vec.clone())?,
+            generics: param_types_no_nested_vec.clone()
+        }
+        .contains_no_nested_vectors());
+        assert!(!ParamType::Enum {
+            name: "EnumName".to_string(),
+            variants: EnumVariants::new(tuples_with_nested_vec)?,
+            generics: param_types_no_nested_vec
+        }
+        .contains_no_nested_vectors());
+        assert!(!ParamType::Enum {
+            name: "EnumName".to_string(),
+            variants: EnumVariants::new(tuples_no_nested_vec)?,
+            generics: param_types_nested_vec
+        }
+        .contains_no_nested_vectors());
+        Ok(())
```

Comment:
Although this might not qualify as a 100% unit test, it might (or might not) be interesting to you:
```suggestion
        assert!(Vec::<u8>::param_type().contains_no_nested_vectors());
        assert!(!Vec::<Vec<u8>>::param_type().contains_no_nested_vectors());

        assert!(<[u8; 10]>::param_type().contains_no_nested_vectors());
        assert!(!<[Vec<u8>; 10]>::param_type().contains_no_nested_vectors());

        assert!(<(u64, u32)>::param_type().contains_no_nested_vectors());
        assert!(!<((), Byte, Vec<u8>)>::param_type().contains_no_nested_vectors());

        #[derive(Parameterize)]
        #[FuelsTypesPath("crate")]
        struct StructCanHoldAVecInGeneric<T: Parameterize> {
            bim: u16,
            bam: bool,
            generic: T,
        }
        assert!(
            !StructCanHoldAVecInGeneric::<((), Byte, Vec<u8>)>::param_type()
                .contains_no_nested_vectors()
        );
        assert!(
            StructCanHoldAVecInGeneric::<((), Byte, u8)>::param_type().contains_no_nested_vectors()
        );

        #[derive(Parameterize)]
        #[FuelsTypesPath("crate")]
        struct StructHasAVecAsAField {
            bim: u16,
            bam: bool,
            the_vec: Vec<u8>,
        }
        assert!(!StructHasAVecAsAField::param_type().contains_no_nested_vectors());

        #[derive(Parameterize)]
        #[FuelsTypesPath("crate")]
        enum EnumCanHoldAVecInGeneric<T: Parameterize> {
            Bim(u16),
            Bam(bool),
            TheGeneric(T),
        }
        assert!(EnumCanHoldAVecInGeneric::<(u64, u32)>::param_type().contains_no_nested_vectors());
        assert!(!EnumCanHoldAVecInGeneric::<Vec<u8>>::param_type().contains_no_nested_vectors());

        #[derive(Parameterize)]
        #[FuelsTypesPath("crate")]
        enum EnumHasAVecAsAVariant {
            Bim(u16),
            Bam(bool),
            TheVec(Vec<u8>),
        }
        assert!(!EnumHasAVecAsAVariant::param_type().contains_no_nested_vectors());

        Ok(())
```

Reply:
Will keep this form in mind for the `Bytes`, I really just want to get `Vec`s out asap